net-utils/src/ip_echo_client.rs

```diff
@@ -226,6 +226,8 @@ pub(crate) async fn verify_all_reachable_tcp(
 /// necessary if checking many ports.
 /// A given amount of retries will be made to accommodate packet loss.
 /// This function may panic.
+///
+#[allow(tail_expr_drop_order)]
```

Comment:
Which specific problem are we masking here? Can this be more narrowly scoped?

Reply:
I guess you could also do a loop here:
```
loop {
   let next = checkers.join_next().await;
   let Some(r) = next else { 
      break; 
   }
}
```
which looks somewhat cleaner